mesonbuild/interpreter/interpreter.py

```diff
@@ -664,6 +664,9 @@ def func_import(self, node: mparser.BaseNode, args: T.Tuple[str],
     @typed_pos_args('files', varargs=str)
     @noKwargs
     def func_files(self, node: mparser.FunctionNode, args: T.Tuple[T.List[str]], kwargs: 'TYPE_kwargs') -> T.List[mesonlib.File]:
+        for filename in args[0]:
+            if filename.endswith(' '):
+                raise MesonException(f'{filename!r} ends with a space. This is probably an error.', file=node.filename, lineno=node.lineno, colno=node.colno)
```

Comment:
Passing file/lineno/colno is *usually* not necessary, since the exception handler has access to the latest node information from the time the exception was raised and updates the exception if none was provided.

It's still useful in case you need fine-grained node information, especially when coding around the parser itself when we haven't yet set up the interpreter's auto-handling of node info. But that's not the common case.

This also means that lines can be a lot shorter. :)